### PR TITLE
STAC-N1 on k8s pods

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
/cc @mrsiano @jeremyeder

### CODE REVIEW

File: networking/synthetic/stac-on-k8s/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+**Pre-requisites:**
+-----------------------
+- K8s 1.8 or higher patched with the [PR for device plugin's extended support for smart-NICs](https://github.com/kubernetes/kubernetes/pull/51938)
+- Onload 1.3+ installed and working on the hosts.
+
+**Steps:**
+------
+
+ **Deploy SFC device plugin daemonset pods:**
```
Comment (mrsiano) [inline networking/synthetic/stac-on-k8s/ReadMe.md]: what about all the export commands ?

File: networking/synthetic/stac-on-k8s/Dockerfile-rhel74-onload [hunk 2]
```diff
@@ -0,0 +1,56 @@
+FROM registry.access.redhat.com/rhel7/rhel
+
+MAINTAINER Vikas Choudhary "vichoudh@redhat.com"
+
+RUN yum install -y --nogpgcheck \
+        python-devel \
```
Comment (mrsiano) [inline networking/synthetic/stac-on-k8s/Dockerfile-rhel74-onload]: please add "tuned" and "python34"

File: networking/synthetic/stac-on-k8s/Dockerfile-rhel74-onload [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    rm -f /lib/systemd/system/local-fs.target.wants/*; \
+    rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
+    rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
+    rm -f /lib/systemd/system/basic.target.wants/*; \
+    rm -f /lib/systemd/system/anaconda.target.wants/*; \
+    rm -f /lib/systemd/system/systemd*udev*; \
+    rm -f /lib/systemd/system/getty.target; \
+    ln -s /usr/lib64/libgmp.so.10 /usr/lib64/libgmp.so.3
+
+RUN wget http://download.eng.pek2.redhat.com/rel-eng/latest-RHEL-7/compose/Server/x86_64/os/Packages/kernel-`uname -r`.rpm \
```
Comment (jeremyeder) [inline networking/synthetic/stac-on-k8s/Dockerfile-rhel74-onload]: hostnamed have to be abstracted out

File: networking/synthetic/stac-on-k8s/Dockerfile-rhel74-onload [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    && wget http://download.eng.pek2.redhat.com/rel-eng/latest-RHEL-7/compose/Server/x86_64/os/Packages/kernel-devel-`uname -r`.rpm \
+    && wget http://download.eng.pek2.redhat.com/rel-eng/latest-RHEL-7/compose/Server-optional/x86_64/os/Packages/libpcap-devel-1.5.3-9.el7.i686.rpm \
+    && yum install -y *.rpm
+
+RUN wget -q http://www.openonload.org/download/openonload-201606-u1.3.tgz \
+    && wget http://perf1.perf.lab.eng.bos.redhat.com/psuriset/vikas/vikas-pstac-n.tar.gz \
```
Comment (jeremyeder) [inline networking/synthetic/stac-on-k8s/Dockerfile-rhel74-onload]: again, private hostnames

File: networking/synthetic/stac-on-k8s/stac-consumer.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: stac-consumer
+  annotations:
+    sfc-nic-ip: 70.70.70.2/24
```
Comment (jeremyeder) [inline networking/synthetic/stac-on-k8s/stac-consumer.yml]: I am wondering if we want something to actually generate these yaml files...ansible to generate the yaml itself.

File: networking/synthetic/stac-on-k8s/stac-producer.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: stac-producer
+  annotations:
+    sfc-nic-ip: 70.70.70.1/24
+spec:
+  containers:
+  - name: stac-producer
+    image: rhel74:latest
```
Comment (jeremyeder) [inline networking/synthetic/stac-on-k8s/stac-producer.yml]: you might want to name this image rhel74 to indicate that onload is in there.

File: networking/synthetic/stac-on-k8s/start-stac-test [hunk 7]
```diff
@@ -0,0 +1,28 @@
+cd /home/openonload-201707-with-vi-availability/scripts
+./onload_install --nobuild --userfiles
+
+
+mv /vikas-pstac-n.tar.gz /home/
```
Comment (jeremyeder) [inline networking/synthetic/stac-on-k8s/start-stac-test]: what is vikas-pstac-n.tar.gz?

File: networking/synthetic/stac-on-k8s/ReadMe.md (hunk 1 above)
Comment (mrsiano) [reply networking/synthetic/stac-on-k8s/ReadMe.md]: I have to run this to successfully deploy the sfc:
"kubectl config set-context $(kubectl config current-context) --namespace=kube-system"

**Comment by mffiedler** [comment]
@vikaschoudhary16 Do you want to update for the review comments?  or should I close this one out?

**Comment by mffiedler** [comment]
Please reopen if and when we are ready to go forward with this.

**Pull request closed without merging**